src/test/java/seedu/address/model/VersionedAddressBookTest.java

```diff
@@ -0,0 +1,104 @@
+package seedu.address.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static seedu.address.model.VersionedAddressBook.MESSAGE_NO_MORE_HISTORY;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalPersons.getTypicalAddressBook;
+
+import java.util.ArrayList;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+
+public class VersionedAddressBookTest {
+    private ModelManager model;
+    private AddressBook addressBook1;
+    private AddressBook addressBook2;
+    private VersionedAddressBook versionedAddressBook;
+
+    @BeforeEach
+    void setUp() {
+        addressBook1 = new AddressBook();
+        model = new ModelManager(addressBook1, new UserPrefs());
+        versionedAddressBook = model.getVersionedAddressBook();
+        addressBook2 = getTypicalAddressBook();
+    }
+
+    @Test
+    void testConstructor() {
+        ArrayList<AddressBook> expectedStateList = new ArrayList<>();
+        expectedStateList.add(addressBook1);
+
+        assertEquals(expectedStateList, versionedAddressBook.getAddressBookStateList());
+        assertEquals(1, versionedAddressBook.getAddressBookStateList().size());
+    }
+
+    @Test
+    void testCommitAddressBook() {
+        model.setAddressBook(addressBook2);
```

Comment:
Sorry for the leftover part! I should change this to resetData instead of using model here. Do you mind if I go ahead and fix it?

Reply:
looks like `model.setAddressBook(addressBook2)` is just calling `versionedAddressBook(addressBook2)` under the hood?